gandalf.js

```diff
@@ -1,118 +1,113 @@
-const randomIndex = Math.floor(Math.random() * array.length);
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.